diff --git a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
index 5c792bd..d070e0e 100644
--- a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
+++ b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
@@ -66,6 +66,7 @@ public class SvgView extends ReactViewGroup implements ReactCompoundView, ReactC
     }
 
     private @Nullable Bitmap mBitmap;
+    private boolean mRemovalTransitionStarted = false;
 
     public SvgView(ReactContext reactContext) {
         super(reactContext);
@@ -90,12 +91,30 @@ public class SvgView extends ReactViewGroup implements ReactCompoundView, ReactC
             ((VirtualView) parent).getSvgView().invalidate();
             return;
         }
-        if (mBitmap != null) {
-            mBitmap.recycle();
+        // Additional: maybe its not necessary since Android 2.3.3 https://developer.android.com/topic/performance/graphics/manage-memory#recycle
+        if(!mRemovalTransitionStarted){
+            if (mBitmap != null) {
+                mBitmap.recycle();
+            }
+            mBitmap = null;
+        }
+    }
+
+    @Override
+    public void startViewTransition(View view) {
+        super.startViewTransition(view);
+        mRemovalTransitionStarted = true;
+    }
+
+    @Override
+    public void endViewTransition(View view) {
+        super.endViewTransition(view);
+        if (mRemovalTransitionStarted) {
+            mRemovalTransitionStarted = false;
         }
-        mBitmap = null;
     }
 
+
     @Override
     protected void onDraw(Canvas canvas) {
         if (getParent() instanceof VirtualView) {